### PR TITLE
Fix: Fix 3dp profile category operation support bugs

### PR DESCRIPTION
**Pull request by MaShizhen**

### CODE REVIEW

File: src/app/ui/views/ProfileManager/index.jsx [hunk 1]
```diff
@@ -130,7 +149,10 @@ function ProfileManager({
                 return;
             }
             if (status) {
-                const title = definitionState.isCategorySelected ? currentDefinition.category : currentDefinition.name;
+                const title = definitionState.isCategorySelected ? (
+                    // Assign default values to import type
+                    currentDefinition.category || i18n._('key-default_category-Custom')
```
Comment (jane-rose) [inline src/app/ui/views/ProfileManager/index.jsx:154]: same here

File: src/app/lib/uni-api.js [hunk 2]
```diff
@@ -262,19 +276,35 @@ const File = {
             if (!targetFile) throw new Error('export file canceled');
 
             const file = { path: targetFile, name: renderGcodeFileName };
-            fs.copyFileSync(tmpFile, targetFile);
+            if (isProfileConfig) {
+                const txt = fs.readFileSync(tmpFile, 'utf-8');
```
Comment (xiekun1992) [inline src/app/lib/uni-api.js:280]: `'utf-8'` -> `'utf8'`

File: src/app/lib/uni-api.js [hunk 3]
```diff
@@ -262,19 +276,35 @@ const File = {
...
-                .get(`/data${tmpFile}`)
-                .responseType('blob')
-                .end((err, res) => {
-                    // FileSaver.saveAs(res.body, targetFile, true);
-                    FileSaver.saveAs(res.body, renderGcodeFileName, true);
-                    callback && callback('web');
-                });
+            if (isProfileConfig) {
+                request
+                    .get(`/data${tmpFile}`)
```
Comment (xiekun1992) [inline src/app/lib/uni-api.js:292]: `/data` is the value of constant variable `DATA_PATH`, use `${DATA_PATH}${tmpFile}`

File: src/app/lib/uni-api.js [hunk 4]
```diff
@@ -262,19 +276,35 @@ const File = {
...
+                    .end((err, res) => {
+                        const json = res.body;
+                        const newProfile = this.resetProfile(json);
+                        FileSaver.saveAs(new Blob([newProfile]), renderGcodeFileName, true);
+                        callback && callback('web');
+                    });
+            } else {
+                request
+                    .get(`/data${tmpFile}`)
```
Comment (xiekun1992) [inline src/app/lib/uni-api.js:301]: `/data` is the value of constant variable `DATA_PATH`, use `${DATA_PATH}${tmpFile}`

**Review by xiekun1992 — CHANGES_REQUESTED** [review]

File: src/app/flux/editor/actions-process.js [hunk 5]
```diff
@@ -524,31 +524,55 @@ export const processActions = {
...
+            // set a new name that cannot be repeated
+            formData.append('uploadName', `${file.name.substr(0, file.name.length - 9)}${timestamp()}.def.json`);
+            api.uploadFile(formData)
+                .then(async (res) => {
+                    const response = res.body;
+                    const definitionId = `New.${timestamp()}`;
+                    const definition = await definitionManager.uploadDefinition(definitionId, response.uploadName);
+
+                    let name = definition.i18nName ? (
+                        i18n._(definition.i18nName) || definition.name
```
Comment (wen-qing-255) [inline src/app/flux/editor/actions-process.js:540]: why need to `|| definition.name` ?

**Review by bbaa77770 — APPROVED** [review]

**Review by bbaa77770 — COMMENTED** [review]
Try to reduce use 'i18nCategory'.

File: src/app/flux/editor/actions-process.js (hunk 5 above)
Comment (MaShizhen) [reply src/app/flux/editor/actions-process.js:540]: Avoid errors due to lack of I18N comparison

Comment (wen-qing-255) [reply src/app/flux/editor/actions-process.js:540]: but i think i18n._() always return value

**Comment by MaShizhen** [comment]
`i18nCategory` has some reading scenes
    - Category exists in profile file of old version, restore through I18N when importing
    - render ProfileManager CateList
    - The category of the exported profile is consistent with the current language

`i18nCategory`Set scene
    - copy default profile，keep the original i18ncategory
    - copy custom profile，clear i18ncategory

File: src/app/flux/editor/actions-process.js (hunk 5 above)
Comment (MaShizhen) [reply src/app/flux/editor/actions-process.js]: done

File: src/app/ui/views/ProfileManager/index.jsx [hunk 6]
```diff
@@ -109,15 +121,22 @@ function ProfileManager({
...
 
     const actions = {
-        isCategorySelectedNow: (category) => {
+        isCategorySelectedNow: (category, i18nCategory) => {
             const { definitionForManager, isCategorySelected } = definitionState;
-            return isCategorySelected && definitionForManager.category === category;
+            if (!isCategorySelected) {
+                return false;
+            }
+            if (i18nCategory === 'key-default_category-Custom') {
```
[the remark this answers is not in the document]
Comment (MaShizhen) [reply src/app/ui/views/ProfileManager/index.jsx]: done

**Review by xiekun1992 — APPROVED** [review]

**Review by jane-rose — APPROVED** [review]

**Pull request merged**